forte/data/extractor/attribute_extractor.py

```diff
@@ -0,0 +1,189 @@
+        add them into the vocabulary.
+
+        Args:
+            pack (DataPack): The datapack that contains the current
+                instance.
+            instance (Annotation): The instance from which the
+                extractor will extractor feature.
+        """
+        for entry in pack.get(self.config.entry_type, instance):
+            element = self.get_attribute(entry, self.config.attribute)
+            if not isinstance(element, Hashable):
+                raise AttributeError("Only hashable element can be"
+                    "added into the vocabulary. Consider setting"
+                    "vocab_method to be raw and do not call update_vocab"
+                    "if you only need the raw attribute value without"
+                    "coverting them into index.")
+            self.add(element)
+
+    def extract(self, pack: DataPack, instance: Annotation) -> Feature:
+        r"""Functionality: Extract attributes of one instance.
+        For example, the text of tokens in one sentence.
+
+        Args:
+            pack (Datapack): The datapack that contains the current
+                instance.
+            instance (Annotation): The instance from which the
+                extractor will extractor feature.
+
+        Returns:
+            Feature: a feature that contains the extracted data.
+        """
+        data = []
+        for entry in pack.get(self.config.entry_type, instance):
+            value = self.get_attribute(entry, self.config.attribute)
+            rep = self.element2repr(value) if self.vocab else value
+            data.append(rep)
+
+        meta_data = {"need_pad": self.config.need_pad,
+                     "pad_value": self.get_pad_value(),
+                     "dim": 1,
+                     "dtype": int if self.vocab else Any}
+
+        return Feature(data=data,
+                       metadata=meta_data,
+                       vocab=self.vocab)
+
+    def pre_evaluation_action(self, pack: DataPack, instance: Annotation):
+        r"""Funcationality: Remove attributes of one instance. For
+        example remove all pos tags of tokens in one sentence, if the
+        entry_type is Token and the attribute is pos.  This function is
+        called before the evaluation on a pack. After features are removed,
+        new features predicted from the model will be added to the pack.
+
+        Args:
+            pack (Datapack): The datapack that contains the current
+                instance.
+            instance (Annotation): The instance from which the
+                extractor will extractor feature.
+        """
+        for entry in pack.get(self.config.entry_type, instance):
+            self.set_attribute(entry, self.config.attribute, None)
+
+    def add_to_pack(self, pack: DataPack, instance: Annotation,
+                    prediction: Iterable[Union[int, Any]]):
+        r"""Functionality: Add the prediction for attribute to the
```

Comment:
You can remove this `Functionality:` start.

Reply:
got it.